File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int cs=0,ms=-(int)(1e9);
        for(int i=0;i<nums.length;i++){
            cs=cs+nums[i];
            if(ms<cs)
               ms=cs;
            if(cs<0)
               cs=0;
        }
        return ms;
    }
}

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int cur = 0;
        int maxi = nums[0];
        int n = nums.size();

        for(int i = 0; i < n; i ++){
            cur += nums[i];
            maxi = max(maxi, cur);
            cur = max(cur, 0);
        }

        return maxi;

        
    }
};